### PR TITLE
feat: add native tags implementation

### PR DESCRIPTION
**Pull request by mxschmitt**
https://github.com/microsoft/playwright/issues/26327

### CODE REVIEW

File: tests/playwright-test/tags.spec.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    `
+  });
+  expect(exitCode).toBe(0);
+  expect(passed).toBe(6);
+});
+
+test('config.grep should work', async ({ runInlineTest }) => {
+  const result = await runInlineTest({
+    'playwright.config.ts': `
+      module.exports = { grep: /@tag1/ };
```
Comment (mxschmitt) [inline tests/playwright-test/tags.spec.ts:55]:
```
  tags: RegExp | string[] | string
  tagsInvert: <same-as-above>
```

p: (multiple strings are `and`, RegExp, is `or`)

File: tests/playwright-test/tags.spec.ts [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+    const result = await runInlineTest({
+      'playwright.config.ts': `
+      module.exports = {};
+    `,
+      'a.test.ts': `
+      import { test, expect } from '@playwright/test';
+      test.tag('tag1')('test1', async () => { console.log('\\n%% test1'); });
+      test('test2', async () => { console.log('\\n%% test2'); });
+    `,
+    }, { 'tag': 'tag1' });
```
Comment (dgozman) [inline tests/playwright-test/tags.spec.ts:326]:
```suggestion
    }, { 'tagInvert': 'tag1' });
```

**Comment by Az8th** [comment]
Sounds really promising, thanks for this addition 🎉 
I have just a little concern about the HTML reporter : removing tags from the test title string may break the search if the regex only applies to it.
I think a filter by tags could be interesting to add, or even a dropdown menu with all detected ones, and checkboxes for each ruling their visibility

**Comment by pavelfeldman** [comment]
Closing as stale

**Pull request closed without merging**